### PR TITLE
add Bpoly.antiderivatives / integrals

### PR DESCRIPTION
**Pull request by ev-br**
Add the functionality for integrating piecewise polynomials directly in the Bernstein basis without converting them to the power basis.

### CODE REVIEW

File: scipy/interpolate/interpolate.py [hunk 1]
```diff
@@ -1142,6 +1143,80 @@ def derivative(self, nu=1):
...
+            Definite integral of the piecewise polynomial over [a, b]
+
+        """
+        # XXX: can probably use instead the fact that
+        # \int_0^{1} B_{j, n}(x) \dx = 1/(n+1)
+        ib = self.antiderivative()
+        if extrapolate is not None:
+            ig.extrapolate = extrapolate
+        return ib(b) - ib(a)
+
```
Comment (argriffing) [inline scipy/interpolate/interpolate.py]: I'm confused about `ig`.  Is the conditional branch not tested?

Comment (argriffing) [inline scipy/interpolate/interpolate.py]: also `ouf-of-bounds`

Comment (ev-br) [inline scipy/interpolate/interpolate.py]: Nice catch, thanks!
Fixed in the last commit.

**Comment by ev-br** [comment]
Fixed the typo --- thanks Alex! (the typo and the comment by @argriffing have disappeared when I force-pushed the last commit)

File: scipy/interpolate/interpolate.py [hunk 2]
```diff
@@ -1142,6 +1144,81 @@ def derivative(self, nu=1):
...
+        Returns
+        -------
+        bp : BPoly
+            Piecewise polynomial of order k2 = k + nu representing the
+            antiderivative of this polynomial.
+
+        """
+        if nu <= 0:
+            raise self.derivative(-nu)
```
Comment (pv) [inline scipy/interpolate/interpolate.py]: return, not raise

File: scipy/interpolate/interpolate.py [hunk 3]
```diff
@@ -1142,6 +1144,81 @@ def derivative(self, nu=1):
...
+
+        # Now fix continuity: on the very first interval, take the integration
+        # constant to be zero; on an interval [x_j, x_{j+1}) with j>0,
+        # the integration constant is then equal to the jump of the `bp` at x_j.
+        # The latter is given by the coefficient of B_{n+1, n+1}
+        # *on the previous interval* (other B. polynomials are zero at the breakpoint)
+        # Finally, use the fact that BPs form a partition of unity.
+        for j in range(1, x.shape[0]-1):
+            x0 = x[j]
+            c2[:, j, ...] += c2[k, j-1, ...]
```
Comment (pv) [inline scipy/interpolate/interpolate.py]: Probably can be vectorized. `c2[:,1:] += cumsum(c2[k,:], axis=0)[:-1]`?

Comment (ev-br) [inline scipy/interpolate/interpolate.py]: Done in the force-pushed version.

**Comment by ev-br** [comment]
I believe I addressed all review comments, all branches seem to be tested.

**Comment by pv** [comment]
Thanks, lgtm

**Comment by pv** [comment]
@ev-br: Ok, I missed the new axis argument. It probably should be passed to construct_fast also here in the antiderivative routine.

File: scipy/interpolate/interpolate.py [hunk 4]
```diff
@@ -1142,6 +1144,79 @@ def derivative(self, nu=1):
...
+
+        # Now fix continuity: on the very first interval, take the integration
+        # constant to be zero; on an interval [x_j, x_{j+1}) with j>0,
+        # the integration constant is then equal to the jump of the `bp` at x_j.
+        # The latter is given by the coefficient of B_{n+1, n+1}
+        # *on the previous interval* (other B. polynomials are zero at the breakpoint)
+        # Finally, use the fact that BPs form a partition of unity.
+        c2[:,1:] += np.cumsum(c2[k,:], axis=0)[:-1]
+
+        return self.construct_fast(c2, x, self.extrapolate)
```
Comment (pv) [inline scipy/interpolate/interpolate.py]: Sorry I missed this one: maybe `self.construct_fast(c2, x, self.interpolate, axis=self.axis)`?

Comment (ev-br) [inline scipy/interpolate/interpolate.py]: Will do. As soon as I get to a real keyboard, that is (am traveling).

**Pull request merged**